web/src/dropdown_widget.ts

```diff
@@ -468,10 +498,34 @@ export class DropdownWidget {
                     this.item_click_callback(event, instance, this, true);
                 });
 
-                // Set focus on first element when dropdown opens.
+                // Adjust focus based on how the dropdown was opened
                 setTimeout(() => {
                     if (this.hide_search_box) {
-                        $dropdown_list_body.find(".list-item:first-child").trigger("focus");
+                        if (this.dropdown_opened_by_keyboard_click) {
+                            const $selected_item = $dropdown_list_body.find(
+                                `.list-item[data-unique-id="${this.current_value}"]`,
+                            );
+                            $selected_item.trigger("focus");
+                        } else {
+                            assert(this.list_widget !== undefined);
+                            const first_item = this.list_widget.get_current_list()[0];
+                            if (first_item) {
+                                const $first_item = $popper.find(
+                                    `.list-item[data-unique-id="${first_item.unique_id}"]`,
+                                );
+                                this.$events_container.one(
+                                    "keydown",
+                                    `${this.widget_selector}, ${this.widget_wrapper_id}`,
+                                    (e) => {
+                                        if (e.key === "ArrowDown" || e.key === "ArrowUp") {
+                                            $first_item.trigger("focus");
+                                            e.stopPropagation();
+                                            e.preventDefault();
+                                        }
+                                    },
+                                );
+                            }
+                        }
```

Comment:
The logic sounds good but it would be nice to have some comments or modify the logic to make this code more readable.